src/network.py

```diff
@@ -265,10 +265,11 @@ def __init__(self, input_dim=3, units_per_layer=(3, 2), acts=('relu', 'sigmoid')
 
             # for every unit in the current layer create layer's units
             for j in range(units_per_layer[i]):
+                #TODO: this part needs to be checked. It overrites the values assigned by weight_init
                 units.append(
-                    Unit(w=weights_inits(type=weights_init, n_weights=n_weights, lower_lim=0., upper_lim=1.,
+                    Unit(w=weights_inits(type=weights_init, n_weights=n_weights, lower_lim=0.1, upper_lim=0.2,
                                          value=weights_value),
-                         b=weights_inits(type=weights_init, n_weights=1, lower_lim=0., upper_lim=1.,
+                         b=weights_inits(type=weights_init, n_weights=1, lower_lim=0., upper_lim=0.01,
```

Comment:
Questa parte qui andava a sovrascrivere i valori che passavamo con weight_init tramite Network.
Per testare valori in range 0.1 - 0.2 (come suggerisce Micheli nelle slides sul progetto) ho dovuto cambiare i valori direttamente da qui.
Un altra roba, il bias veniva settato con valori tra 0 e 1. Quando il bias ha valori alti (>0.5) questo influisce molto sul training, ed i risultati sono un pò instabili. Secondo me conviene settare il bias a 0 oppure con valori vicini a 0.
Impostandolo manualmente                        
```suggestion
                         b=weights_inits(type=weights_init, n_weights=1, lower_lim=0., upper_lim=0.01,
```
i risultati sul training sono nettamente migliori. 

Un altra cosa, sulle slides di Micheli dice avoid:
- all zeros
- high values
- all equal valus 
A questo punto un inizializzazione uniform sembra inutile. Che dici?